cadence/freshmint-claim-sale/FreshmintClaimSale.cdc

```diff
@@ -260,14 +310,16 @@ pub contract FreshmintClaimSale {
         access(self) fun checkAllowlist(address: Address) {
             if let allowlistCap = self.allowlist {
 
-                let allowlist = allowlistCap.borrow() ?? panic("failed to borrow allowlist")
+                let allowlist = allowlistCap.borrow() 
+                    ?? panic("failed to borrow allowlist")
 
                 if let claims = allowlist.getClaims(address: address) {
                     if claims == 0 {
                         panic("address has already consumed all claims")
                     }
 
-                    allowlist.consumeClaim(address: address)
+                    // Reduce the claim count by one
+                    allowlist.setClaims(address: address, claims: claims - 1)
```

Comment:
Kinda curious on why the function was removed 👀

Reply:
I realized that `consumeClaim` was redundant. It reads from the `claimsByAddress` dictionary to get the claim count, but we already do that in this function. Removing it means we only read from the dictionary once per claim, which is more performant.